injector/cpu_pressure.go

```diff
@@ -90,83 +75,106 @@ func (i *cpuPressureInjector) Inject() error {
-			i.config.Log.Infow("joining target cpuset cgroup", "core", core, "pid", pid)
+	i.config.Log.Infow("all routines have been created successfully, now stressing", "stresserPIDPerCore", i.config.StresserManager.GetStresserPIDs())
 
-			if err = i.config.Cgroup.Join("cpuset", pid, false); err != nil {
-				i.config.Log.Errorw("failed to join the target cpuset cgroup", "error", err, "core", core, "pid", pid)
+	return nil
+}
 
-				return
-			}
+func (i *cpuPressureInjector) joinCgroup(core int, pid int) bool {
+	// join target CPU cgroup
+	i.config.Log.Infow("joining target CPU cgroup", "core", core, "pid", pid)
 
-			// prioritize the current process
-			i.config.Log.Infow("highering current process priority", "core", core, "pid", pid)
+	if err := i.config.Cgroup.Join("cpu", pid, false); err != nil {
+		i.config.Log.Errorw("failed join the target CPU cgroup", "error", err, "core", core, "pid", pid)
+		return false
+	}
 
-			if err = i.config.ProcessManager.Prioritize(); err != nil {
-				i.config.Log.Errorw("error highering the current process priority", "error", err, "core", core, "pid", pid)
+	// join target cpuset cgroup in case it is used to pin the target on specific cores
+	i.config.Log.Infow("joining target cpuset cgroup", "core", core, "pid", pid)
 
-				return
-			}
+	if err := i.config.Cgroup.Join("cpuset", pid, false); err != nil {
+		i.config.Log.Errorw("failed to join the target cpuset cgroup", "error", err, "core", core, "pid", pid)
+		return false
+	}
 
-			i.config.Log.Infow("starting the stresser", "core", core, "pid", pid)
+	return true
+}
 
-			mutex.Lock()
-			tids = append(tids, pid)
-			i.routines++
-			mutex.Unlock()
+func (i *cpuPressureInjector) startStresser(core int, pid int) {
+	i.config.Log.Infow("starting the stresser", "core", core, "pid", pid)
+	i.config.StresserManager.TrackStresser(core, pid)
+	i.config.Stresser.Stress(i.config.StresserExit)
+}
 
-			wg.Done()
-			i.config.Stresser.Stress(i.config.StresserExit)
-		}(core)
+func (i *cpuPressureInjector) prioritizeStresserProcess(core int, pid int) bool {
+	i.config.Log.Infow("prioritizing the current stresser process", "core", core, "pid", pid)
+
+	if err := i.config.ProcessManager.Prioritize(); err != nil {
+		i.config.Log.Errorw("failed to prioritize the current stresser process", "error", err, "core", core, "pid", pid)
+		return false
 	}
 
-	// wait for stress routines to finish initializing
-	wg.Wait()
+	return true
+}
 
-	if !succeeded {
-		return fmt.Errorf("at least one stresser routine failed to execute")
+func (i *cpuPressureInjector) detectTargetCores() cpuset.CPUSet {
+	if err := i.config.StresserManager.TrackInjectorCores(i.config); err != nil {
+		i.config.Log.Errorw("failed to parse CPUSet", "error", err)
 	}
 
-	i.config.Log.Infow("all routines have been created successfully, now stressing", "routinesPID", tids)
+	cores := i.config.StresserManager.GetCoresToBeStressed()
```

Comment:
Here we are returning all cores up to our containers.
This seems counter intuitive to me.
We still want for an injector config, the associated cores. Not the whole list of cores IMO.

I understand this works, however it feels misleading to me and error prone.

Reply:
I get it. Let me rework to do 2 things
1) Get `injectorCores` (to be stressed) from the config
2) TrackInjectors should union and set GOMAXPROCS with the whole list of cores

Reply:
Yes this seems definitely where the GOMAXPROCS should be changed, once the amount of cores to be stressed changes 👍